javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,98 @@
+let userName = null; 
+let todoList = []; 
+
+function getReply(command) {
+  const lowerCaseCommand = command.toLowerCase();
+
+  if (lowerCaseCommand.startsWith('hello my name is ')) {
+    const name = command.slice(17); 
+    if (userName === name) {
+      return `You already introduced yourself as ${name}.`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${name}`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what is my name?') {
+    return userName
+      ? `Your name is ${userName}`
+      : "I don't know your name yet. Please tell me your name first.";
+  }
+
+  if (
+    lowerCaseCommand.startsWith('add ') &&
+    lowerCaseCommand.includes(' to my todo')
+  ) {
+    const todo = command.slice(4, command.indexOf(' to my todo'));
+    todoList.push(todo);
+    return `${todo} added to your todo.`;
+  }
+
+  if (
+    lowerCaseCommand.startsWith('remove ') &&
+    lowerCaseCommand.includes(' from my todo')
+  ) {
+    const todo = command.slice(7, command.indexOf(' from my todo'));
+    const index = todoList.indexOf(todo);
+    if (index > -1) {
+      todoList.splice(index, 1); 
+      return `Removed ${todo} from your todo.`;
+    } else {
+      return `${todo} is not in your todo list.`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what is on my todo?') {
+    if (todoList.length === 0) {
+      return 'Your todo list is empty.';
+    } else {
+      return `You have ${todoList.length} todos: ${todoList.join(', ')}.`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what day is it today?') {
+    const today = new Date();
+    const day = today.getDate();
+    const month = today.toLocaleString('default', { month: 'long' }); 
+    const year = today.getFullYear();
+    return `${day}. of ${month} ${year}`;
+  }
+
+  if (lowerCaseCommand.startsWith('what is ')) {
+    const mathExpression = command.slice(8); 
+    try {
+      const result = eval(mathExpression); 
```

Comment:
While `eval()` is very useful for cases like this where we want to evaluate an expression, we need to be very careful about using it as developers.

Many sites will disable using it entirely these days, but consider what happens if the user types
```
what is alert(`hello ${userName}`)
```

That's right - this user input will cause the browser (if running in the browser) to pop up an alert dialog, and it's even got access to the value of any variables! Consider that this could be used to send information about the user to a malicious site, and you'll start to realise why `eval()` is generally considered best left alone!

That said, for this assignment, it's a great way to get started.

An alternative could be to parse some numbers and an operator (+, -, *, /) form the command string, and then perform a different operation based on the operator.